cmd/healthcheck-handler.go

```diff
@@ -27,22 +27,44 @@ import (
 func ReadinessCheckHandler(w http.ResponseWriter, r *http.Request) {
 	ctx := newContext(r, w, "ReadinessCheckHandler")
 
-	objLayer := newObjectLayerWithoutSafeModeFn()
-	// Service not initialized yet
-	if objLayer == nil {
-		writeResponse(w, http.StatusServiceUnavailable, nil, mimeNone)
-		return
+	ready := func() bool {
+		objLayer := newObjectLayerWithoutSafeModeFn()
+		// Service not initialized yet
+		if objLayer == nil {
+			return false
+		}
+
+		ctx, cancel := context.WithTimeout(ctx, globalAPIConfig.getReadyDeadline())
+		defer cancel()
+		if !objLayer.IsReady(ctx) && newObjectLayerFn() == nil {
```

Comment:
can be written in a single line

```suggestion
		return objLayer.IsReady(ctx)
```